### PR TITLE
Prevent overwriting form model value with default value

### PR DESCRIPTION
**Pull request by alymdrictels**
Covers IntegerHandler, Checkbox and Float field types

Complete coverage for DataMapper.mapFormsToData

### CODE REVIEW

File: Tests/Form/DataMapper/CreateContentMapperTest.php [hunk 1]
```diff
@@ -521,4 +521,17 @@ public function testMapFormsToDataWhenDataIsNotDataWrapperSecondCase()
...
+    {
+        $someNumber = 42;
+
+        try {
+            $this->mapper->mapFormsToData(array(), $someNumber);
+        } catch (\Symfony\Component\Form\Exception\UnexpectedTypeException $e) {
+            // empty body
+        }
+
+        $this->assertInstanceOf('\Symfony\Component\Form\Exception\UnexpectedTypeException',$e);
```
Comment (pspanja) [inline Tests/Form/DataMapper/CreateContentMapperTest.php]: Use `@expectedException` annotation in the docblock instead.

File: Tests/Form/FieldTypeHandler/CheckboxTest.php [hunk 2]
```diff
@@ -5,62 +5,235 @@
...
     }
 
+
     public function testBuildFieldCreateForm()
     {
-        $formBuilder = $this->getMockBuilder(FormBuilder::class)
-            ->disableOriginalConstructor()
-            ->setMethods(array('add'))
-            ->getMock();
+        $formBuilder = clone($this->formBuilder);
```
Comment (pspanja) [inline Tests/Form/FieldTypeHandler/CheckboxTest.php]: Why cloning? `setUp()` should be automatically called before each test, so this should not be needed.

File: Form/FieldTypeHandler/IntegerHandler.php [hunk 3]
```diff
@@ -23,8 +41,14 @@ protected function buildFieldForm(
     ) {
         $options = $this->getDefaultFieldOptions($fieldDefinition, $languageCode, $content);
 
-        if (!empty($fieldDefinition->defaultValue)) {
-            $options['data'] = (int)$fieldDefinition->defaultValue->value;
+        if ($fieldDefinition->defaultValue instanceof IntegerValue\Value) {
+            if ($content instanceof Content) {
+                if ($this->fieldHelper->isFieldEmpty($content, $fieldDefinition->identifier, $languageCode)) {
+                    $options['data'] = (int)$fieldDefinition->defaultValue->value;
```
Comment (pspanja) [inline Form/FieldTypeHandler/IntegerHandler.php]: This is the case for updating, right (`$content` being set)? Why do you set the default value if the field is empty in this case? AFAICS it should never be set for update.

File: Tests/Form/FieldTypeHandler/CheckboxTest.php [hunk 4]
```diff
@@ -5,62 +5,235 @@
...
+            )
+        );
+
+        $formBuilder = clone($this->formBuilder);
+        $formBuilder->expects($this->once())
+            ->method('add')->withConsecutive(array(
+                $fieldDefinition->identifier, 'checkbox', $options
+            ));
+
+        $checkboxHandler = new Checkbox($this->fieldHelper);
```
Comment (pspanja) [inline Tests/Form/FieldTypeHandler/CheckboxTest.php:165]: Missing assert that `$fieldHelper::isEmpty()` is never called.

**Comment by pspanja** [comment]
+1, ping @MarioBlazek

**Comment by MarioBlazek** [comment]
Looks good, +1

**Comment by pspanja** [comment]
Great, thanks @alymdrictels, merging

**Pull request merged**